### DISCUSSION TITLE
DLL ordinals, .lib and .exp files

### DISCUSSION THREAD

**Discussion by fenugrec** [NationalSecurityAgency/ghidra#7097] [Q&A]
Hi, I'm working on a winCE 4 binary that imports DLL functions by ordinals, say from COREDLL.DLL, MFCCE400.DLL. The latter is fine since the CE SDK includes a mfcce400.pdb which I was eventually able to process and use. However, for coredll.dll, the SDK only has a .lib and .exp file, which AFAIK should be sufficient to match ordinals to function names, but I'm not sure how to get that information into ghidra.

- importing the .lib as I would a .dll isn't really a solution : the .lib being a container file, I don't see how to process the whole thing. Loading in "batch mode" is useless , "filesystem mode" is impractical (it lists hundreds of unnamed components which I could select manually).
- importing the .exp file : ghidra identifies it as a COFF file with the correct architecture, but doesn't find exports or any symbols.

**Comment by ryanmkurtz** [comment]
My knowledge is currently very rusty here, but try running the `CreateExportFileForDLL.java` GhidraScript with your analyzed DLL open in Ghidra.

**Reply by fenugrec** [reply]
thanks, I actually did that for mfcce400.dll , but in the case of coredll ... I haven't been able to find that file yet !

**Comment by fenugrec** [comment]
I eventually found a compatible version of `coredll.dll`, with its PDB (that currently cannot be loaded however, c.f. #7100 ) , so this became an non-issue, but I'm still interested to know if there's way to extract the same information from the .exp or .lib files.
